includes/class-migration.php

```diff
@@ -146,6 +147,9 @@ public static function maybe_migrate() {
 		if ( \version_compare( $version_from_db, '4.1.0', '<' ) ) {
 			self::migrate_to_4_1_0();
 		}
+		if ( \version_compare( $version_from_db, '4.5.0', '<' ) ) {
```

Comment:
If you move this block to the `async_migration` function, it will be handled async!

Reply:
you don't have to!

Reply:
ah, these are batches...

Reply:
I always love to have something that works for all implementations... should be add a third param for various data, so that all async jobs can add additional data?

Reply:
I knew you would say that 😄
Let me take another look at it

Reply:
With the batching I'm not sure how to make it work in a reasonable way. We'd have to carry forward the old version, re-triggering any upgrade routines in `async_migration`, depending on which version the plugin upgraded from.